### PR TITLE
Adds eclipse ide setup guide to docs along with recent code style file and some mkdocs fixes. Fixes #1149

### LINKED ISSUE #1149: Add documentation on how to get the project working in Eclipse
# Task Description

Most of us are using IntelliJ where there are no extra settings required. Developers using Eclipse seem to need more steps in order to configure things in their IDE. We should add a section (or, if necessary, a separate page) for this in the [`strongbox-docs`](https://github.com/strongbox/strongbox-docs/).

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

### PR DESCRIPTION
**Pull request by kalski**
This PR fixes [#1149](https://github.com/strongbox/strongbox/issues/1149)
* Added eclipse ide setup guide
* Updated eclipse code style
* Updated mkdocs dependencies and their description

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -28,8 +28,10 @@ plugins:
 
 extra:
   url: &url !!python/object/apply:os.getenv ["DEPLOY_URL", "http://localhost:8000"]
-  assets: &assets !!python/object/apply:os.path.join [*url, "assets"]
-  resources: !!python/object/apply:os.path.join [*assets, "resources"]
+  _url_path: &_url_path !!python/object/apply:urllib.parse.urljoin [*url, "./"]
+  assets: &assets !!python/object/apply:urllib.parse.urljoin [*_url_path, "assets"]
+  _assets_path: &_assets_path !!python/object/apply:urllib.parse.urljoin [*_url_path, "assets/"]
+  resources: !!python/object/apply:urllib.parse.urljoin [*_assets_path, "resources"]
```
Comment (steve-todorov) [inline mkdocs.yml]: Could you please change the variables to:
```suggestion
extra:
  url: &url !!python/object/apply:os.getenv ["DEPLOY_URL", "http://localhost:8000"]
  assets: !!python/object/apply:urllib.parse.urljoin [*url, "/assets"]
  resources: !!python/object/apply:urllib.parse.urljoin [*url, "/assets/resources"]
```
It is easier to read and understand (and I can live without referencing the `assets` variable in the `resources`) :)

File: docs/developer-guide/ide-setup/eclipse.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Eclipse
+
+!!! tip "Before continuing, please make sure you've built the code using [Building / Building the code / Building strongbox] section."
```
Comment (steve-todorov) [inline docs/developer-guide/ide-setup/eclipse.md]: Cosmetic: Could you please change `[Building / Building the code / Building strongbox]` to `[Building the code]` ?

File: docs/developer-guide/ide-setup/eclipse.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        ![strongbox-aql -> Properties -> Java Build Path -> Add Library -> Groovy Runtime Libraries][AQL groovy project nature]
+    * `Project Natures -> Add -> Groovy Nature`
+        ![strongbox-aql -> Properties -> Project Natures -> Add -> Confirm Project Nature update -> Groovy Nature][AQL groovy libraries in Build Path]
+
+### Trigger maven update for `strongbox-masterbuild`
+Select `strongbox-masterbuild` project, Right click -> Maven -> Update Project
+
+![strongbox-masterbuild -> Maven -> Update Project][masterbuild maven update]
+
+[Building / Building the code / Building strongbox]: ../developer-guide/building-the-code.html#building-strongbox
```
Comment (steve-todorov) [inline docs/developer-guide/ide-setup/eclipse.md]: Cosmetic: This should also be renamed to `[Building the code]`. Also, the link should be `../building-the-code.md#building-strongbox`

**Review by steve-todorov — COMMENTED** [review]
@kalski I've left a few comments (mainly cosmetic things). Also, could you please move the screenshots into `screenshots/ide/eclipse-setup/**`? I will move the Strongbox related screenshots into `screenshots/strongbox` in another PR.   

Thanks for taking the time and writing this guide! :)

File: mkdocs.yml (hunk 1 above)
Comment (kalski) [reply mkdocs.yml]: @steve-todorov I should have added comment on why these were necessary. 

But I agree your suggestion makes more sense and I'll clean these up.

File: docs/developer-guide/ide-setup/eclipse.md (hunk 3 above)
Comment (kalski) [reply docs/developer-guide/ide-setup/eclipse.md]: Thanks for spotting this! I forgot to fix this one when I moved the guide to the proper place.

Comment (kalski) [reply docs/developer-guide/ide-setup/eclipse.md]: I'll also document [here](https://github.com/strongbox/strongbox-docs/pull/12/files#diff-04c6e90faac2675aa89e2176d2eec7d8L43) on the mkdocs suggested way to refer internal links.

File: docs/developer-guide/ide-setup/eclipse.md (hunk 2 above)
Comment (kalski) [reply docs/developer-guide/ide-setup/eclipse.md]: I was just following the path navigation but I'll change it, sure.

**Review by steve-todorov — APPROVED** [review]
LGTM. Many thanks for all your efforts! :)

**Pull request merged**